imf/imf.py

```diff
@@ -678,38 +678,63 @@ def __init__(self,mmin,mmax,gamma,sigma):
         self.gamma = gamma
 
     def __call__(self,m, integral_form=False):
+        m = np.asarray(m)
         if integral_form:
             #       Returns
             #       -------
             #       Probability that m < x for the given CDF with specified mmin,mmax,sigma, and gamma
-            if self.mmax<self.mmin:
-                raise ValueError("Mmax must be greater than Mmin")
-            def error(t):
+            
+                if self.mmax<self.mmin:
+                    raise ValueError("U must be greater than L")
```

Comment:
this check should be done once before the `if integral_form` check, rather than once here and once in the `else` below.